### DISCUSSION TITLE
Gunner Gameplay - Advanced Gunnery system

### DISCUSSION THREAD

**Discussion by Alkanty** [PeasantFoundry/Mechtrauma#45] [Ideas]
**Vanilla Gameplay :** 
Vanilla gameplay around subs guns evolves around : 

- Fire gun that spin around a certain ° real fast
- Reload weapons with ammo 
- repair when weapon station break 

Why change it ?  

- too repetitive gameplay loop (fire-repair-reload)
- Gameplay can be too easy with advanced weapon against vanilla mob 
- can't create new weapon because nearly everything is overpowered once mastered. 

**Real life exemple**
You have different exemple of how certains "tank" or "heavy gunner turret" on ship operate but based on documentations  : 
- You operate turret, you move it around based on motor and hydrolics  that makes the turret spin
- you maintain it, barrel wise, things can get stuck into it, malfunction, repetitive use with heat dilemna. 
- you AIM, based on coordination of an operator, or plain sight if the target is close by.
- You reload ammo, often manually (some systems are automatics like the Leclerc tank for exemple)  
- you engage target or not based of the nature of the target, distance, condition, etc... 
- You use shell based on targets and materials (HEAT, AP... ) 
- you coordinate with your other crew : some turrets are mostly automatic now with the gunner on the deck with the other operators (ships) or within the same space as the gun system (tank)
![image](https://user-images.githubusercontent.com/83227201/216631495-68679ea6-2491-4611-a06c-f5e0d28b7e29.png)
_a warship gun-turret_

**What could enchanced the gameplay?**

- Turrets use motors to spin around, they break at impact and by use, once broken the turret axis is stuck and need to be replaced outside of the ship (making reperation outside dangerous but necessary) _Easy to implement ? use mechtrauma mechannics_
- Some Monsters have real resistance of certain type of ammunition (carapace, skin, size, etc..) making the change of ammo relevant 
_moderate to implement ? can monster have ammo type resistance  in baro ?_
- Reduce line of sight of the gunner by a  cone shape when in turret, can only see thru a tunnel sight and thats it. The tunnel sight can be broken like the motor, by impact or by use, leaving the player essentialy with a dark screen, that has to be guided by sound (sonar ?), or by operator _hard to implement ? change aim sight of turret, make a fog of war around ?_
![Barotrauma jpg bdb159c1412691519f684256a17e5397](https://user-images.githubusercontent.com/83227201/216674189-f1076673-427c-4e44-b6e3-9c32ab372aeb.jpg)
-Make sound system or sonar system for monster so they can be detected other than by visual (horror aspect because bipping sound is more scary than visual sight). Or an other way around is to put a screen UI like the captain one that spots monsters into the gunner view.  _hard to make and maybe not necessary/ UI model_
- make a compass model for both gunner and captain so you can give indication to your gunner _Hard to make/UI model_
- Some weapons may jam, electronics can fall, and that implies reloading the gun with switch, button(like UI from the reactor for exemple) and takes time
- A jauge with heating capacity of some weapons, laser would be  suceptible of heating more than other and start fires ! 
- The turret rotation speed can be reduced and the firing skill too, but the power of one bullet can be more impactfull (more skill related)
- Friendly fire on ship,
- Other tool to take of monster that do not implies only turret should be buffed (electrical discharge, Decoil, etc..)   
-upgrade for better rotor speed (motor change in turret) or line of sight (better gun periscope) 
-some monster can affect ship navigation in blurring sonar, line of sight, IEM. corosive mob that has to be dealt with special ammunition (harpoon gun ? ) 
- More impactfull audio for rotation of turret, Turret are heavy and fire huge caliber, the ship need to feel it ! 
-Ship shaking and moving at fire big gun
-Energy consumption of big gun need to be raised so that it become a choice to fire those guns that can lead to good or bad situation for the ship. 

Voila, if you like any idea in this list and want me to elaborate on certain points (stats, etc...) i am willing to help ! Unfortunatly i am not a programmer so i can't help for this aspect, (i am just starting to learn Python scripting for server maintenance so i am not there yet ;) ). This is a great mod you are providing the community ! 

PS : Sorry for bad english, not native

**Comment by TBN-MapleWheels** [comment]
> - Turrets use motors to spin around, they break at impact and by use, once broken the turret axis is stuck and need to be replaced outside of the ship (making reperation outside dangerous but necessary) Easy to implement ? use mechtrauma mechannics

Motors for servo components (ie. a turret) is a nice idea. 

> - Some Monsters have real resistance of certain type of ammunition (carapace, skin, size, etc..) making the change of ammo relevant moderate to implement ? can monster have ammo type resistance in baro ?

Yes, this is already a thing in vanilla, it's just not very obvious to the player.

> - Reduce line of sight of the gunner by a cone shape when in turret, can only see thru a tunnel sight and thats it. The tunnel sight can be broken like the motor, by impact or by use, leaving the player essentialy with a dark screen, that has to be guided by sound (sonar ?), or by operator hard to implement ? change aim sight of turret, make a fog of war around ?
> - Make sound system or sonar system for monster so they can be detected other than by visual (horror aspect because bipping sound is more scary than visual sight). Or an other way around is to put a screen UI like the captain one that spots monsters into the gunner view. hard to make and maybe not necessary/ UI model make a compass model for both gunner and captain so you can give indication to your gunner Hard to make/UI model

Sonar-based aiming is already a thing. No one uses it because aiming normally is easier. This is a sub design choice, not a mod choice.

> Some weapons may jam, electronics can fall, and that implies reloading the gun with switch, button(like UI from the reactor for exemple) and takes time

This would just annoy players. I would say this needs to be tied to weapon maintenance. 

> - A jauge with heating capacity of some weapons, laser would be suceptible of heating more than other and start fires ! The turret rotation speed can be reduced and the firing skill too, but the power of one bullet can be more impactfull (more skill related)

This is weapon-dependent. Maybe a consideration.

> - Friendly fire on ship

Already a thing in vanilla.

> - upgrade for better rotor speed (motor change in turret) or line of sight (better gun periscope)

Yeah, this is possible.

> - some monster can affect ship navigation in blurring sonar, line of sight, IEM. corosive mob that has to be dealt with special ammunition (harpoon gun ? )

Already a thing in vanilla. Again, no one uses Sonar aiming because it's too convenient to use the periscope.

> - More impactful audio for rotation of turret, Turret are heavy and fire huge caliber, the ship need to feel it !
> - Ship shaking and moving at fire big gun

One of the reasons I like EK Gunnery turrets but this might exceed the scope of Mechtrauma.

> - Energy consumption of big gun need to be raised so that it become a choice to fire those guns that can lead to good or bad situation for the ship.

Already a thing in vanilla, this is also controlled by the sub designer. Not something for Mechtrauma to address.

---
Overall, I like the idea of turrets having components that wear down. I think other things outside the sub should also need maintenance. This would force the player to leave the sub which can create good moments of horror and intensity.

The points from the above I personally like would be :
- The turret and other outside components needing repairable/consumable component items.
- Weapon jamming for **poorly** maintained turrets.
- Upgradeable components as an alternative to the generic turret upgrades offered by the Submarine Engineer Merchant at outposts.
